src/wordpress-playground/index.tsx

```diff
@@ -297,3 +300,98 @@ function zipNameToHumanName(zipName) {
 }
 
 main();
+
+async function overwriteFile() {
+	const targetFile = document.getElementById('target-file').value;
+	const targetContent = document.getElementById('target-content').value;
+	console.log(targetFile);
+	console.log(targetContent);
+	await workerThread.writeFile(targetFile, targetContent);
+}
+
+async function populateZip(zipFile, filePaths) {
+	for (let i = 0; i < filePaths.length; i++) {
+		const path = filePaths[i];
+		const isDir = await workerThread.isDir(path);
+		if (isDir) {
+			const directoryFileNames = await workerThread.listFiles(path);
+			const directoryFilePaths = directoryFileNames.map(
+				(childFilePath) => `${path}/${childFilePath}`
+			);
+			await populateZip(zipFile, directoryFilePaths);
+		} else {
+			const fileContents = await workerThread.readFile(path);
+			zipFile.file(path, fileContents);
+		}
+	}
+}
+
+async function generateZip() {
```

Comment:
PHP is already compiled with libzip support. Using `ZipArchive` would remove the need for the `JSZip` dependency and would also defer the heavy lifting to the worker thread. I'm not sure if it would be slower or faster than JSZip, but could be worth a shot.

Reply:
The code has been revised to remove dependency on JSZip.